app.py

```
@@ -148,352 +298,645 @@
-        print("Showing reports for all time.") # Debugging
-        # For 'all time', date_filter_sql remains empty and semester_query_params is empty
+        selected_semester = 'all' # Ensure 'all' is the value if invalid or default
+        print("No semester filter applied.")
 
     try:
-        # --- Build queries with optional date filtering (for overall and trend) ---
-        # These only filter by semester/time, not course
-        overall_sql = f"SELECT SUM(present), COUNT(*) FROM attendance a {date_filter_sql.replace(' WHERE', 'WHERE') if date_filter_sql else ''}"
-        cursor.execute(overall_sql, semester_query_params) # Use semester_query_params
+        # --- Overall Attendance Percentage (All Time or Semester) ---
+        overall_sql = f"SELECT SUM(present), COUNT(*) FROM attendance a {date_filter_sql_overall_trend.replace(' WHERE', 'WHERE') if date_filter_sql_overall_trend else ''}"
```

Ruff (0.8.2)

Possible SQL injection vector through string-based query construction

(S608)